sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py

```diff
@@ -428,6 +438,66 @@ def _update_inputs(self, pipeline_inputs):
                 anno.description = self._args_description.get(input_name)
                 self.inputs[input_name] = anno
 
+    @classmethod
+    def _get_output_annotation(cls, func):
+        """Get the output annotation of the function, validate & refine it."""
+        return_annotation = inspect.signature(func).return_annotation
+
+        if is_group(return_annotation):
+            outputs = _get_param_with_standard_annotation(return_annotation, is_func=False)
+        elif isinstance(return_annotation, Output):
+            outputs = {cls.DEFAULT_OUTPUT_NAME: return_annotation}
+        else:
+            # skip if return annotation is not group or output
+            return {}
+
+        output_annotations = {}
+        for key, val in outputs.items():
+            if isinstance(val, GroupInput):
```

Comment:
Will we rename this GroupInput class in the future? it's a little weird GroupInput exists in outputs.